src/v/cloud_storage/async_manifest_view.cc

```diff
@@ -506,23 +512,31 @@ async_manifest_view::get_cursor(
               to_string(query));
             co_return error_outcome::out_of_range;
         }
+        model::offset end = _stm_manifest.get_last_offset();
+        if (cursor_range) {
+            end = cursor_range->end_inclusive;
+        }
+
         model::offset begin;
-        model::offset end = end_inclusive.value_or(
-          _stm_manifest.get_last_offset());
         if (_stm_manifest.get_archive_start_offset() == model::offset{}) {
+            // Archive not initialised
             begin = _stm_manifest.get_start_offset().value_or(begin);
+        } else if (cursor_range && cursor_range->start_inclusive) {
+            // Start offset overriden
+            begin = cursor_range->start_inclusive.value();
         } else {
+            // Default of archive start offset
             begin = _stm_manifest.get_archive_start_offset();
```

Comment:
here we should probably just use `get_archive_clean_offset` and that's it

Reply:
It's a good idea and it would make things a lot simpler. I'm a bit concerned about the read path racing with the archive garbage collection, but in that case we should just return an error, so maybe it's fine.

Reply:
Updated with this approach. Thanks for the suggestion!